console.* functions called outside tests are not output when singleRun is false

### Expected behaviour
Using `client.captureConsole = true`, I expect all `console.*` calls to output to the terminal.
```
d:\karma_captureConsole_bug> karma start
Chrome 68.0.3440 (Windows 10 0.0.0) INFO: 'I am outside of test!'

LOG: 'I am inside of test!'
Chrome 68.0.3440 (Windows 10 0.0.0): Executed 1 of 1 SUCCESS (0.007 secs / 0.001 secs)
TOTAL: 1 SUCCESS
```

### Actual behaviour
Only `console.*` calls occurring while running a test are output to the terminal
```
d:\karma_captureConsole_bug> karma start
LOG: 'I am inside of test!'
Chrome 68.0.3440 (Windows 10 0.0.0): Executed 1 of 1 SUCCESS (0.007 secs / 0.001 secs)
TOTAL: 1 SUCCESS
```

### Environment Details

- Karma version (output of `karma --version`): `3.0.0`

#### `karma.conf.js`:
```javascript
module.exports = (config) => {
  config.set({
    client: {captureConsole: true},
    frameworks: ['jasmine', 'requirejs'],
    files: ['karma.test.js', 'script.js'],
    browsers: ['Chrome'],
    singleRun: false,
  })
};
```
#### `karma.test.js`:
```javascript
require.config({
  deps: ['script'],
  callback: window.__karma__.start
});
```
#### `script.js`:
```javascript
define('script', [], function(){
  console.info('I am outside of test!');
  it('test1', function(){
    console.log('I am inside of test!');
  });
});
```

### Steps to reproduce the behaviour

1. put above `karma.conf.js`, `karma.test.js`, `script.js` in a directory
2. run `karma start` to see actual behavior
3. change `karma.conf.js` so that `singleRun` is `true` to observe expected behavior

Thanks, I tried the repo but
```
 ./node_modules/karma/bin/karma start
07 11 2018 07:13:18.826:ERROR [karma-server]: Server start failed on port 9876: Error: Cannot find module 'mocha'
```

--- a/lib/browser.js
+++ b/lib/browser.js
@@ -48,25 +48,23 @@ class Browser {
   onKarmaError (error) {
     if (this.isNotConnected()) {
       this.lastResult.error = true
-      this.emitter.emit('browser_error', this, error)
-      this.refreshNoActivityTimeout()
     }
+    this.emitter.emit('browser_error', this, error)
+    this.refreshNoActivityTimeout()
   }
 
   onInfo (info) {
-    if (this.isNotConnected()) {
-      if (helper.isDefined(info.dump)) {
-        this.emitter.emit('browser_log', this, info.dump, 'dump')
-      }
-
-      if (helper.isDefined(info.log)) {
-        this.emitter.emit('browser_log', this, info.log, info.type)
-      } else if (!helper.isDefined(info.dump)) {
-        this.emitter.emit('browser_info', this, info)
-      }
-
-      this.refreshNoActivityTimeout()
+    if (helper.isDefined(info.dump)) {
+      this.emitter.emit('browser_log', this, info.dump, 'dump')
     }
+
+    if (helper.isDefined(info.log)) {
+      this.emitter.emit('browser_log', this, info.log, info.type)
+    } else if (!helper.isDefined(info.dump)) {
+      this.emitter.emit('browser_info', this, info)
+    }
+
+    this.refreshNoActivityTimeout()
   }
 
   onStart (info) {

--- a/test/e2e/browser_console.feature
+++ b/test/e2e/browser_console.feature
@@ -172,3 +172,39 @@ Feature: Browser Console Configuration
       .
       HeadlessChrome
       """
+  Scenario: Execute logging program with singleRun
+    Given a configuration with:
+      """
+      files = ['browser-console/log.js', 'browser-console/test.js'];
+      browsers = ['ChromeHeadlessNoSandbox'];
+      plugins = [
+        'karma-jasmine',
+        'karma-chrome-launcher'
+      ];
+      singleRun = false;
+      """
+    When I runOut Karma
+    Then it passes with like:
+      """
+      LOG: 'foo'
+      """
+    Then it passes with like:
+      """
+      DEBUG: 'bar'
+      """
+    Then it passes with like:
+      """
+      INFO: 'baz'
+      """
+    Then it passes with like:
+      """
+      WARN: 'foobar'
+      """
+    Then it passes with like:
+      """
+      ERROR: 'barbaz'
+      """
+    Then it passes with like:
+      """
+      SUCCESS
+      """

--- a/test/e2e/error.feature
+++ b/test/e2e/error.feature
@@ -18,3 +18,21 @@ Feature: Error Display
       """
       SyntaxError: Unexpected token }
       """
+  Scenario: Single-run Syntax Error in a test file
+    Given a configuration with:
+      """
+      files = ['error/test.js', 'error/under-test.js'];
+      browsers = ['ChromeHeadlessNoSandbox'];
+      plugins = [
+        'karma-jasmine',
+        'karma-chrome-launcher'
+      ];
+      singleRun = false;
+      """
+    When I monitor Karma
+    And I stop when the log contains 'SyntaxError'
+    Then it fails with like:
+      """
+      SyntaxError: Unexpected token }
+      """
+    And I stop a server programmatically

--- a/test/e2e/step_definitions/core_steps.js
+++ b/test/e2e/step_definitions/core_steps.js
@@ -64,7 +64,7 @@ cucumber.defineSupportCode((a) => {
         }
 
         const runOut = command === 'runOut'
-        if (command === 'run' || command === 'runOut') {
+        if (command === 'run' || command === 'runOut' || command === 'monitor') {
           this.child = spawn('' + runtimePath, ['start', '--log-level', 'warn', configFile])
           const done = () => {
             cleansingNeeded = true
@@ -96,6 +96,9 @@ cucumber.defineSupportCode((a) => {
                 }
                 done()
               })
+              if (command === 'monitor') {
+                done()
+              }
             }, 1000)
           })
         } else {
@@ -157,7 +160,7 @@ cucumber.defineSupportCode((a) => {
 
   defineParameterType({
     name: 'command',
-    regexp: /run|runOut|start|init|stop/
+    regexp: /run|runOut|start|init|stop|monitor/
   })
 
   defineParameterType({
@@ -175,6 +178,15 @@ cucumber.defineSupportCode((a) => {
 
   When('I {command} Karma behind a proxy on port {int} that prepends {string} to the base path', function (command, proxyPort, proxyPath, callback) {
     execKarma.apply(this, [command, undefined, proxyPort, proxyPath, callback])
+  })
+
+  When('I stop when the log contains {string}', function (message, callback) {
+    setInterval(() => {
+      if (this.lastRun.stdout.includes(message)) {
+        this.child && this.child.kill()
+        callback()
+      }
+    }, 100)
   })
 
   defineParameterType({

--- a/test/unit/browser.spec.js
+++ b/test/unit/browser.spec.js
@@ -91,14 +91,11 @@ describe('Browser', () => {
       expect(spy).to.have.been.called
     })
 
-    it('should ignore if browser not executing', () => {
-      const spy = sinon.spy()
-      emitter.on('browser_error', spy)
+    it('should not set lastResult if browser not executing', () => {
       browser.state = Browser.STATE_CONNECTED
 
       browser.onKarmaError()
       expect(browser.lastResult.error).to.equal(false)
-      expect(spy).not.to.have.been.called
     })
   })
 

